apps/checkout-app/frontend/utils.ts

```diff
@@ -43,13 +45,62 @@ export const unflattenSettings = <T, S extends Node>(
   return unflattenedSettings;
 };
 
+/**
+ * Merges settings. To be used when default and saved settings may differ (e.g. after app update).
+ * @param defaultSettings
+ * @param savedSettings
+ * @returns Returns either previously saved settings values or default settings values. If settings values are present in default and saved at the same time, then saved value is returned.
+ */
+const mergeSettingsValues = (
```

Comment:
This could be made bit simpler (without nested reduce)
```suggestion
const mergeSettingsValues = (
  defaultSettings: UnknownSettingsValues,
  savedSettings: UnknownSettingsValues
) => {
  return reduce(
    defaultSettings,
    (result, defaultSetting, settingKey) => {
      const savedSetting = savedSettings[settingKey];

      const hasSettingInBothSettings = !!savedSetting;

      const udpatedSetting = hasSettingInBothSettings
        ? { ...defaultSetting, ...savedSetting }
        : defaultSetting;

      return {
        ...result,
        [settingKey]: udpatedSetting,
      };
    },
    savedSettings
  );
};
```

Reply:
I added tests and updated function

Reply:
.